coalib/results/result_actions/IgnoreResultAction.py

```diff
@@ -46,22 +47,32 @@ def apply(self, result, original_file_dict, file_diff_dict, language: str,
         """
         Add (I)gnore comment
         """
+        source_range = next(filter(lambda sr: exists(sr.file),
+                                   result.affected_code))
+        filename = source_range.file
+
+        original_line = original_file_dict[filename][
+                        source_range.start.line-1].rstrip()
 
-        ignore_comment = self.get_ignore_comment(result.origin, language)
+        ignore_comment = self.get_ignore_comment(result.origin, language,
+                                                 original_line)
 
         if not ignore_comment:
             return file_diff_dict
 
-        source_range = next(filter(lambda sr: exists(sr.file),
-                                   result.affected_code))
-        filename = source_range.file
-
         ignore_diff = Diff(original_file_dict[filename])
-        ignore_diff.change_line(
-            source_range.start.line,
-            original_file_dict[filename][source_range.start.line-1],
-            original_file_dict[filename][source_range.start.line-1].rstrip() +
-            '  ' + ignore_comment)
+
+        # Add the new ignore using 'and' with the previous ignore.
+        if ignore_comment == result.origin + ' and':
+            changed_line = re.sub(r'(Ignore\s)(\S+)', r'\1{} \2'.format(
```

Comment:
If 2 or more ignore comments are already present in the line, then it would make the comment look like - `Ignore ThisBear and ThatBear and ThatBear`. Don't you think it be better if the comment looked like -- `Ignore ThisBear, ThatBear and ThatBear`?

Reply:
This was discussed here before: https://github.com/coala/coala/pull/5069#discussion_r161372051

I wonder why it wasn't added :thinking:   cc @pareksha

Reply:
I had discussed this with @Makman2 on coala/coala thread, he said 'and' would be alright as either we have to put ``,`` or ``and``. It will look like either ``Ignore ThisBear and ThatBear and ThatBear`` or ``Ignore ThisBear, ThatBear, ThatBear``. It can't be ``Ignore ThisBear, ThatBear and ThatBear`` as noone knows whether it's the last bear or there are yet more to come.

Reply:
Yes we don't know if its the last bear, so how about making the method behave a bit more intelligently -- initially it might have `Ignore ThisBear and ThatBear` and then when another bear comes, the message might be changed to `Ignore ThisBear, ThatBear and YetAnotherBear`. What do you think @pareksha ?

Reply:
This is a naive approach that I'm suggesting but maybe we can count the number of ignores and according to that add combinations of `,` and `and`(s). For e.g. if just one ignore is present then we don't need a comma and just add an `and`. Otherwise replace `and` with `,` and add an `and` before the last ignore.

This might not be the optimal solution, but it might lead us to a better solution.

Reply:
I am not sure if counting the number of *ignore*s would work as we want the new ignore comment to have just one `Ignore` word in the comment. That being said, I think we may do something on similar lines and check for the absence/presence of an `and` in the comment. 
If present, we can replace it with a `,` and add an `and` to the end. 
Else, we can just add an `and` to the end.